### ISSUE TITLE
Update of other packages cancelled when a package fails the validity check

### ISSUE THREAD

**Issue by willemw12** [rmarquis/pacaur#311]
Maybe it is not a bug, however I have noticed that when one package fails the validity check, then the upgrade of the other (top-level) packages is cancelled. Expected pacaur to upgrade the other packages.

    $ pacaur -u
    ...
    AUR Packages  (3): bomi-0.9.2-3  gatotray-2.3-1  google-chrome-dev-42.0.2305.3-1 
    ...
    :: Checking gatotray integrity...
    ==> Making package: gatotray 2.3-1 (Thu Feb 19 11:00:16 CET 2015)
    ==> WARNING: Skipping dependency checks.
    ==> Retrieving sources...
      -> Found 4218cadee76b8a9e9e01bb78de2866cdeafb7a5f.tar.gz
    ==> Validating source files with md5sums...
        4218cadee76b8a9e9e01bb78de2866cdeafb7a5f.tar.gz ... FAILED
    ==> ERROR: One or more files did not pass the validity check!
    ...
    :: failed to verify gatotray integrity

bomi and  google-chrome-dev are not upgraded, except if you do `pacaur --ignore gatotray`.

**Comment by rmarquis** [comment]
This is done on purpose. Pacaur should not ignore packages of the behalf of the user.

This behavior also applies to all packages in the dependency chain. In older versions of pacaur, the build would fails in the middle of the transaction, while it now stops right before starting the build process.

**Comment by Jimi-James** [comment]
Can we please have an option that says to ignore these packages? I use some AUR packages that *always* try to upgrade because their version is "latest", and they often fail to upgrade for some new issue that'll be figured out in a month or so, and I don't care about missing out on updates until that gets fixed. With the current setup, I have to try upgrading once, memorize which packages failed, and upgrade again with an --ignore for those packages. That is extremely cumbersome. I need an option that just says "I don't care that they failed; please install the others".

**Comment by rmarquis** [comment]
You've got the `--ignore` flag, just make use of it.

**Comment by Jimi-James** [comment]
I *just* explained why having to use the --ignore flag is cumbersome and doubles the amount of time it takes me to upgrade my packages. Are you suggesting I ignore those more problematic packages all the time and never update them? Because that isn't a solution. Or that I arbitrarily stop ignoring them once in a while, hoping it'll work? Because that would be just as cumbersome and annoying, and who knows if I would ever even remember to do that. The best solution is to be able to just attempt to upgrade them every time, without having to redo the upgrade to get the others that are ready to go. I don't see a reason that that can't be a non-default option.

**Comment by rmarquis** [comment]
Let me rephrase it for you then: I don't care if pacaur fails if you ignore the reason it fails and want to bypass it blindly. This is not what pacaur is for, and I won't spent time fixing an issue that isn't one.

Your alternative is to send a patch, fork and fix by yourself, or send a significant amount of money to make me implement a workaround for you (see man page for paypal link).

**Comment by Jimi-James** [comment]
I'm not bypassing it blindly. I want to bypass packages that have *nothing* to do with the other packages in the upgrade, besides that they were being upgraded in the same run. No dependency relationships or anything. That saves a lot of time in a lot of situations. In a Bash script, if you're doing a batch operation, the *default* is to ignore failures and keep going, and you have to set the e flag to change that. Why should behavior that is often that is often the *default* be so undesirable that it isn't even an option?

And I can't use another helper, and I'm sorry, but this *is* what pacaur is for. I use pacaur because it's the only AUR helper that can check your package cache for already-built AUR packages, which is a godsend for anyone who maintains more than one Arch install (as I do). Having that feature makes it, to me, the AUR helper that is best for upgrading a system with no headaches. This issue I'm complaining about massively hampers its status as that, and I already still use yaourt for everything but upgrades.

If you won't fix the issue, I will, but just because you don't personally see the point doesn't mean it isn't an issue.

**Comment by rmarquis** [comment]
And I still don't care. One design principle on which pacaur has been built since day one is the [following](https://bbs.archlinux.org/viewtopic.php?pid=937423#p937423):

> By design, pacaur doesn't allow to bypass "issues" that should be fixed upstream in the AUR or in the installed software. Each user should participate to the improvement of the PKGBUILDs by reporting issues to their respective maintainer, instead of selfishly avoiding the issue just for themselves. If you need some manual tinkering, you can simply use cower and the manual way, or you can share and maintain your own modified PKGBUILD in the AUR.

Looks like pacaur is not the right tool for you. You are of course still free to adapt the code for your specific needs. If you require that I spend more of my free time on this, I can do it, but don't expect me to do it for free.

**Comment by Jimi-James** [comment]
Again, what I'm asking is *not* bypassing an issue that should be fixed upstream. If one package fails to download, and another completely unrelated package DOES NOT fail, and they just happen to be in the same batch, there is no bypassing going on by installing the successful one. Bypassing the failure would be trying to installed that package that failed to download anyway. Disagreeing with me means you think that all batch actions on a computer should automatically give up and stop the moment a single member of the batch fails, with no option whatsoever to change that behavior, and many projects including Bash disagree with that statement.

If you don't care, fine, I'll do it myself, but you're being a real asshole about this whole thing. Especially with your refusal to even point out where in the code I'd want to start off the top of your head, without spending more than a few seconds of your free time thinking about it. You're the one who made the program and so can do that. You don't have to refuse to do that and you don't have to inexplicably think that common sense functionality goes against design principles.

**Comment by rmarquis** [comment]
If you can't refrain yourself to insult people, please avoid to comment on these tickets - I'm not your damn slave. Seriously, you didn't even try to have a look at the obvious update code and start backward from there before asking for my free time.

And yes, I absolutely can refuse to work for _free_.

**Comment by Jimi-James** [comment]
I'm starting to see where the misunderstanding is coming from. The very moment you said you didn't want to do it for free, I said I'll do it. I've repeatedly said I'll do it, because you keep reiterating that you won't do it for free. I have *not* been asking you to do it for me at all since my first comment. I've been complaining about your negative and disrespectful tone that comes from being wrong about what you thought I was trying to say. The only thing I asked for since your first "won't spend time" statement was *specifically* any tips you might have off the top of your head, without devoting any free time.

**Comment by rmarquis** [comment]
Again insulting? Guess what, your whole rant isn't worth my limited time. Have a holiday, and a happy fork.

**Comment by mox-mox** [comment]
Wheee, came here as I was trying pacaur for the first time and it failed updating the AUR packages saying:
```
==> ERROR: Failure while downloading ftp://ftp.debian.org/debian/pool/main/c/console-setup/bdf2psf_1.156_all.deb
    Aborting...
:: failed to verify qtox-git integrity
:: failed to verify bdf2psf integrity
```

Some notes:
- Saying that the bdf2psf could not be verified is --while surely true-- misleading. That thing is plain not findable where it is supposed to be anymore.
- For me, there were quite some packages in that update. If there was an option to be asked to skip a failing package on the fly, that would make the task of system administration easier. That way, I can keep the workflow and update the rest of my system before trying to fix the failing ones.
- I have modified the PKGBUILD of qtox-git to use my fork of that project instead of the original one. The first time using pacaur, I was given the chance to edit the PKGBUILD. Now, the second (and following) time, it does not present this option. I can use the --edit option, but then I'll have to re-read all previously approved PKGBUILDs. If pacaur gave an option to only edit the pkgbuilds of failing packages, this would be easier.

Now, all those things are minor hindrances that I could well live with, or maybe try to 'fix' and create a pull request for, if they have a chance of being accepted (Keeping non-trivial forks of software alive is often more cumbersome than living with (perceived) deficits) or consider paying money to have them 'fixed' ('fix' because they are really more of a feature request).
However, there are two questions in my head after reading this thread and the man page of pacaur:
- Is this the general way of treating users of pacaur? I have seen the section in the man page[1].
- Do I want to invest time (either developing or just getting used to a software) or money (and therefore become in a way dependant on it) in a product/software where this is the way users are treated?

I can totally understand if a developer says "Hey, I don't consider this an important problem. If you feel different, please fix it yourself or pay me to do it (if dev is willing to work for money)." I would appreciate it if said developer went on with something like "If I remember correctly, <file/function> would be a good starting point to work on the problem" or "But I have no idea where you could start of, though". At this point, I would have the information, that work on the problem has a chance of being merged and I would have a starting point or know at least that that information is not readily available in somebody elses head. Then, I'd look at the source and decide, if I want to work on the problem or not. These responses would take a developer around two minutes[2] and keep everybody happy. 
Instead, a user asks for a feature, shows, that he/she has researched and understood the provided possibilities, provides a somewhat clear description what the requested feature should be like and why it would improve the product. And gets flamed. Why?

It is completely acceptable to not want to fix/change a program. But after reading the first 10 posts, I get the feeling, that rmarquis did not even care enough to try to understand the request. Yes, calling the option "I don't care that they failed; please install the others" may not be the best name, but still. I do not agree with insulting people over that, but the original request is in no way so abstruse as to warrant such a treatment as shown here. At least two (and counting me late-arrival, three) people do care for it. Is that developer hybris (_I_ know what is good and right), elitism (only MY use case are valid) or some other problem?
If I start using/working-on pacaur and find a problem/bug in a year, will I receive the same treatment?

Best regards,
mox



[1]
```
       If you feel great pity for the present maintainer that has to constantly cope with users that don't want to
       read man pages and refuse to take responsibility for their own system, you can send me funny cat pictures.
       Alternatively, you can send a donation via PayPal to the above email address.

```
[2] Looking at the Issues section, there seem to be no more than around one issue every day.

**Comment by rmarquis** [comment]
> .. as I was trying pacaur for the first time ..
And already complaining. Wow.

> If I start using/working-on pacaur and find a problem/bug in a year, will I receive the same treatment?
Yes. Please fork, create your own helper, or use another one.

**Comment by mox-mox** [comment]
You really don't get it, do you?

I don't really know how you perceive the world and do stuff. Me, when I create something, I'm fond of perfecting it, so I like to have feedback. For you, this does not seem to be true. At least to me, it seems like you treat everything as a personal insult. I have to admit, I feel a bit sad for you, but I'll stick with your advice and not use anything you create. Have a nice day.

**Comment by rmarquis** [comment]
Glad to see you go away! Have a nice day too.

**Comment by se1exin** [comment]
@rmarquis Came by this thread cause I am having a few validity issues on install as well. I've got nothing to report per-se (the issues are obviously package related - not pacaur), but after reading the thread I just wanna give some love and say love your work man, thanks for pacaur! - I know most don't realise the amount of work goes into making and maintaining this kind of thing, but I appreciate it. Thought you could do to hear some positive feedback ;)

**Comment by rmarquis** [comment]
@se1exin Thank you! I really, really appreciate your words. Also thanks for the donation! :)
